--- gst-omx-1.2.0_tunnel/omx/gstomxvideoenc.c.orig	2015-09-09 21:10:37.230446153 -0500
+++ gst-omx-1.2.0_tunnel/omx/gstomxvideoenc.c	2015-09-09 21:10:46.206446587 -0500
@@ -756,8 +756,14 @@
       (guint) buf->omx_buf->nFlags, (guint64) buf->omx_buf->nTimeStamp);
 
   GST_VIDEO_ENCODER_STREAM_LOCK (self);
-  frame = gst_omx_video_find_nearest_frame (buf,
-      gst_video_encoder_get_frames (GST_VIDEO_ENCODER (self)));
+  if (gst_omx_port_is_tunneled (self->enc_in_port)
+      && !(buf->omx_buf->nFlags & OMX_BUFFERFLAG_SYNCFRAME)) {
+    frame = NULL;
+  }
+  else {
+    frame = gst_omx_video_find_nearest_frame (buf,
+        gst_video_encoder_get_frames (GST_VIDEO_ENCODER (self)));
+  }
 
   g_assert (klass->handle_output_frame);
   flow_ret = klass->handle_output_frame (self, self->enc_out_port, buf, frame);
